whelktool/scripts/cleanups/2019/04/remove-duplicated/from-list-move-items-and-delete-instance-duplicates.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * This script moves holdings from one instance to another specified by
+ * the file id_lxl-2281_eboksberikning.txt The holdings are moved from
+ * the instance in the right column to the instance in the left column.
+ *
+ * See LXL-2380 for more info.
+ *
+ */
+
+PrintWriter failedBibIDs = getReportWriter("failed-to-delete-bibIDs")
+PrintWriter scheduledForChange = getReportWriter("scheduledForChange")
+PrintWriter instances = getReportWriter("instances")
+
+def file = new File(scriptDir, "id_lxl-2281_eboksberikning.txt");
+
+file.eachLine { line ->
+    def (goodInstanceId, badInstanceId) = line.split(";").collect { it.trim() }
+
+    def fullGoodInstanceId = "http://libris.kb.se/bib/${goodInstanceId}" as String
+
+    //Create a map of sigels mapping to items held by the good instance
+    def sigelToGoodHoldingMap = [:]
+
+    selectByIds([fullGoodInstanceId]) { goodBib ->
+        instances.println("Good bib ${fullGoodInstanceId} is a ${goodBib.graph[1][TYPE]}")
+        selectBySqlWhere("""
```

Comment:
Adding `collection = 'hold'` to the `WHERE` clause would make this a bit more efficient.